pkg/kubectl/rolling_updater.go

```diff
@@ -67,16 +59,17 @@ type RollingUpdaterConfig struct {
 	// CleanupPolicy defines the cleanup action to take after the deployment is
 	// complete.
 	CleanupPolicy RollingUpdaterCleanupPolicy
-	// UpdateAcceptor is given a chance to accept the new controller after each
-	// scale-up operation. If the controller is accepted, updates continue; if
-	// the controller is rejected, the update will fail immediately.
-	UpdateAcceptor UpdateAcceptor
-	// UpdatePercent is optional; if specified, the amount of replicas scaled up
-	// and down each interval will be computed as a percentage of the desired
-	// replicas for the new RC. If UpdatePercent is nil, one replica will be
-	// scaled up and down each interval. If UpdatePercent is negative, the order
-	// of scaling will be down/up instead of up/down.
+	// UpdatePercent is an optional percentage of replicas to scale up each
+	// interval and is used to compute the minimum pod availability during the
+	// update. If nil, a fixed value of 1 will be used for the scaling increment
+	// and minimum pod availabity. If a negative value is provided, the scaling
+	// direction will be down/up instead of up/down.
 	UpdatePercent *int
+	// UseReadinessAsAcceptance means that the updater will only count ready
+	// pods when determining whether the amount of pods available meets any
+	// minimum pod availability requirement. The default is false, which means
+	// that if a pod exists, it's considered available.
+	UseReadinessAsAcceptance bool
```

Comment:
Possibly if you had an A/B deployment where B is excess capacity, and all
you care about is rollout.

On Aug 6, 2015, at 10:28 PM, Brian Grant <notifications@github.com> wrote:

In pkg/kubectl/rolling_updater.go
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11942#discussion_r36461506>
:

>  	UpdatePercent *int
> +	// UseReadinessAsAcceptance means that the updater will only count ready
> +	// pods when determining whether the amount of pods available meets any
> +	// minimum pod availability requirement. The default is false, which means
> +	// that if a pod exists, it's considered available.
> +	UseReadinessAsAcceptance bool

I'm not 100% opposed to a knob, but (a) readiness should be taken into
account by default and (b) we should understand what the reasons would be
for not using it during rolling update.

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11942/files#r36461506>
.